src/EnergyPlus/RoomAirModelAirflowNetwork.cc

```diff
@@ -0,0 +1,1487 @@
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Brent Griffith
+		//       DATE WRITTEN   November 2009
+		//       MODIFIED
+		//       RE-ENGINEERED
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// calculate new values for temperature and humidity ratio for room air node
+
+		// METHODOLOGY EMPLOYED:
+		// take terms(updated in init routine) and use classic air balance equations
+		// solved for state variables.Store results in structure.
+
+		// REFERENCES:
+		// na
+
+		// Using/Aliasing
+		using DataHVACGlobals::UseZoneTimeStepHistory;
+		using DataHVACGlobals::TimeStepSys;
+		using DataHeatBalFanSys::ZoneVolCapMultpSens;
+		using DataHeatBalFanSys::ZoneVolCapMultpMoist;
+		using DataGlobals::SecInHour;
+		using Psychrometrics::PsyHgAirFnWTdb;
+		using DataHeatBalFanSys::ZoneAirHumRat;
+		using DataHeatBalFanSys::MAT;
+		using DataEnvironment::OutBaroPress;
+		using Psychrometrics::PsyRhFnTdbWPb;
+
+		// Locals
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+		// na
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+		// na
+
+		// DERIVED TYPE DEFINITIONS:
+		// na
+
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+		//unused    INTEGER    :: thisZoneInfo
+		Real64 H2OHtOfVap;
+		Real64 HumRatTmp;
+		Real64 NodeTempX1;
+		Real64 NodeTempX2;
+		Real64 NodeTempX3;
+		Real64 NodeHumRatX1;
+		Real64 NodeHumRatX2;
+		Real64 NodeHumRatX3;
+		Real64 TempDepCoef;
+		Real64 TempIndCoef;
+		Real64 AirCap;
+		Real64 TempTmp;
+		Real64 A;
+		Real64 B;
+		Real64 C;
+		Real64 AirTempT1;
+		Real64 HumRatW1;
+
+		auto & ThisRAFNNode( RoomAirflowNetworkZoneInfo( ZoneNum ).Node( RoomAirNode ) );
+
+		if (UseZoneTimeStepHistory) {
+			NodeTempX1 = ThisRAFNNode.AirTempX1;
+			NodeTempX2 = ThisRAFNNode.AirTempX2;
+			NodeTempX3 = ThisRAFNNode.AirTempX3;
+
+			NodeHumRatX1 = ThisRAFNNode.HumRatX1;
+			NodeHumRatX2 = ThisRAFNNode.HumRatX2;
+			NodeHumRatX3 = ThisRAFNNode.HumRatX3;
+		} 
+		else {  // use down - stepped history
+			NodeTempX1 = ThisRAFNNode.AirTempDSX1;
+			NodeTempX2 = ThisRAFNNode.AirTempDSX2;
+			NodeTempX3 = ThisRAFNNode.AirTempDSX3;
+
+			NodeHumRatX1 = ThisRAFNNode.HumRatDSX1;
+			NodeHumRatX2 = ThisRAFNNode.HumRatDSX2;
+			NodeHumRatX3 = ThisRAFNNode.HumRatDSX3;
+		}
+
+		if ( ZoneAirSolutionAlgo != Use3rdOrder ) {
+			AirTempT1 = ThisRAFNNode.AirTempT1;
+			HumRatW1 = ThisRAFNNode.HumRatW1;
+		}
+		// solve for node drybulb temperature
+		TempDepCoef = ThisRAFNNode.SumHA + ThisRAFNNode.SumLinkMCp + ThisRAFNNode.SumSysMCp;
+		TempIndCoef = ThisRAFNNode.SumIntSensibleGain + ThisRAFNNode.SumHATsurf - ThisRAFNNode.SumHATref + ThisRAFNNode.SumLinkMCpT + ThisRAFNNode.SumSysMCpT + ThisRAFNNode.NonAirSystemResponse + ThisRAFNNode.SysDepZoneLoadsLagged;
+		AirCap = ThisRAFNNode.AirVolume * ZoneVolCapMultpSens * ThisRAFNNode.RhoAir * ThisRAFNNode.CpAir / ( TimeStepSys*SecInHour );
+
+		{ auto const SELECT_CASE_var( ZoneAirSolutionAlgo );
```

Comment:
@lgu1234 The warning about TempTmp is in here. We should change this into a switch statement with a default. Is there a default value for ZoneAirSolutionAlgo?